File: AdvancedTypes/src/main/java/com/bennyhuo/kotlin/advancedtypes/dataclasses/java/Person.java
package com.bennyhuo.kotlin.advancedtypes.dataclasses.java;

public class Person {
    private long id;
    private String name;
    private int age;

    public long getId() {
        return id;
    }

    public Person id(long id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public Person name(String name) {
        this.name = name;
        return this;
    }

    public int getAge() {
        return age;
    }

    public Person age(int age) {
        this.age = age;
        return this;
    }
}

File: Types/src/main/java/com/bennyhuo/kotlin/types/classes/kotlin/Person.kt
package com.bennyhuo.kotlin.types.classes.kotlin

class Person(age: Int, name: String) {
    var age: Int = age //property
        get() {
            return field
        }
        set(value) {
            println("setAge: $value")
            field = value
        }
    var name: String = name
        get() {
            return field // backing field
        }
        set(value) {

        }
}

fun main() {
    val ageRef = Person::age
    val person = Person(18, "Bennyhuo")
    val nameRef = person::name
    ageRef.set(person, 20)
    nameRef.set("Andyhuo")
}